src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -14,9 +15,9 @@ namespace NuGet.Protocol
     /// </summary>
     public class MetadataReferenceCache
     {
-        private readonly Dictionary<string, string> _stringCache = new Dictionary<string, string>(StringComparer.Ordinal);
```

Comment:
I want to see perf measurement with [.NET Core's sleet feed](https://github.com/dotnet/fsharp/blob/de5dc3b8bf6375ef27907ba4a5bb5440360742b1/NuGet.config#L9). If I'm not mistaking ConcurrentDictionary still uses internal locking so I expect size of ConcurrentDictionary increases speed starts lagging.

Reply:
The parallelization being introduced here deals with multi page scenarios and sleet doesn't paginate.

The metadata cache is used in the search scenarios though, so you're right, the search processing *might* be affected here.